pilot/pkg/model/authorization_test.go

```diff
@@ -295,15 +297,143 @@ func TestGetBindingsInNamespace(t *testing.T) {
 		t.Run(tc.name, func(t *testing.T) {
 			authzPolicies := createFakeAuthorizationPolicies(tc.configs, t)
 
-			got := authzPolicies.GetBindingsInNamespace(tc.ns)
+			got := authzPolicies.ListServiceRoleBindings(tc.ns)
 			if !reflect.DeepEqual(tc.want, got) {
 				t.Errorf("want: %v\n but got: %v", tc.want, got)
 			}
 		})
 	}
 }
 
-func TestIsV1RbacEnabled(t *testing.T) {
+func TestAuthorizationPolicies_ListAuthorizationPolicies(t *testing.T) {
+	policy := &authpb.AuthorizationPolicy{
+		Rules: []*authpb.Rule{
+			{
+				From: []*authpb.Rule_From{
+					{
+						Source: &authpb.Source{
+							Principals: []string{"sleep"},
+						},
+					},
+				},
+				To: []*authpb.Rule_To{
+					{
+						Operation: &authpb.Operation{
+							Methods: []string{"GET"},
+						},
+					},
+				},
+			},
+		},
+	}
+	policyWithSelector := proto.Clone(policy).(*authpb.AuthorizationPolicy)
+	policyWithSelector.Selector = &selectorpb.WorkloadSelector{
+		MatchLabels: map[string]string{
+			"app":     "httpbin",
+			"version": "v1",
+		},
+	}
+
+	cases := []struct {
+		name           string
+		ns             string
+		workloadLabels map[string]string
+		configs        []Config
+		want           []Config
+	}{
+		{
+			name: "no policies",
+			ns:   "foo",
+			want: nil,
+		},
+		{
+			name: "no policies in namespace foo",
+			ns:   "foo",
+			configs: []Config{
+				newConfig("authz-1", "bar", policy),
+				newConfig("authz-2", "bar", policy),
+			},
+			want: nil,
+		},
+		{
+			name: "one policy",
+			ns:   "bar",
+			configs: []Config{
+				newConfig("authz-1", "bar", policy),
+			},
+			want: []Config{
+				newConfig("authz-1", "bar", policy),
+			},
+		},
+		{
+			name: "two policies",
+			ns:   "bar",
+			configs: []Config{
+				newConfig("authz-1", "foo", policy),
+				newConfig("authz-1", "bar", policy),
+				newConfig("authz-2", "bar", policy),
+			},
+			want: []Config{
+				newConfig("authz-1", "bar", policy),
+				newConfig("authz-2", "bar", policy),
+			},
+		},
+		{
+			name: "selector exact match",
+			ns:   "bar",
+			workloadLabels: map[string]string{
+				"app":     "httpbin",
+				"version": "v1",
+			},
+			configs: []Config{
+				newConfig("authz-1", "bar", policyWithSelector),
+			},
+			want: []Config{
+				newConfig("authz-1", "bar", policyWithSelector),
+			},
+		},
+		{
+			name: "selector subset match",
+			ns:   "bar",
+			workloadLabels: map[string]string{
+				"app":     "httpbin",
+				"version": "v1",
+				"env":     "dev",
+			},
+			configs: []Config{
+				newConfig("authz-1", "bar", policyWithSelector),
+			},
+			want: []Config{
+				newConfig("authz-1", "bar", policyWithSelector),
+			},
+		},
+		{
+			name: "selector not match",
```

Comment:
Can you add a test that selector match on labels but not on namespace.

Reply:
Good point.